From 6d249217ec4a2d834b83d1334616122bca763c56 Mon Sep 17 00:00:00 2001
From: Sveinung Kvilhaugsvik <sveinung84@users.sourceforge.net>
Date: Mon, 3 Aug 2015 21:39:41 +0200
Subject: [PATCH 8/8] Development version save game compatibility for action
 less unit orders

Load unit orders from 3.0 development version save games that were stored
before the action_list field was added.
---
 server/savegame3.c | 12 ++++++++++--
 1 file changed, 10 insertions(+), 2 deletions(-)

diff --git a/server/savegame3.c b/server/savegame3.c
index 22f3851..e0e26aa 100644
--- a/server/savegame3.c
+++ b/server/savegame3.c
@@ -5091,7 +5091,10 @@ static bool sg_load_player_unit(struct loaddata *loading,
         struct unit_order *order = &punit->orders.list[j];
 
         if (orders_unitstr[j] == '\0' || dir_unitstr[j] == '\0'
-            || act_unitstr[j] == '\0' || action_unitstr[j] == '\0') {
+#ifndef FREECIV_DEV_SAVE_COMPAT
+            || action_unitstr[j] == '\0'
+#endif /* FREECIV_DEV_SAVE_COMPAT */
+            || act_unitstr[j] == '\0') {
           log_sg("Invalid unit orders.");
           free_unit_orders(punit);
           break;
@@ -5100,7 +5103,12 @@ static bool sg_load_player_unit(struct loaddata *loading,
         order->dir = char2dir(dir_unitstr[j]);
         order->activity = char2activity(act_unitstr[j]);
 
-        order->action = (action_unitstr[j] == '?'
+        order->action = (
+#ifdef FREECIV_DEV_SAVE_COMPAT
+                         action_unitstr[0] == '\0'
+                         ||
+#endif /* FREECIV_DEV_SAVE_COMPAT */
+                         action_unitstr[j] == '?'
                          ? ACTION_COUNT
                          : char2num(action_unitstr[j]));
 
-- 
2.1.4

